File: app/src/main/java/com/example/android/miniweather/Network/WeatherService.java
package com.example.android.miniweather.Network;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class WeatherService {

    private final String BASEURL = "http://api.apixu.com/";

    public WeatherApi getService() {
        Retrofit retrofit = new Retrofit.Builder()
                .baseUrl(BASEURL)
                .addConverterFactory(GsonConverterFactory.create())
                .build();

        return retrofit.create(WeatherApi.class);
    }
}

File: app/src/main/java/com/example/android/miniweather/Network/WeatherService.kt
package com.example.android.miniweather.Network

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

class WeatherService {

    private val BASEURL = "http://api.apixu.com/"

    val service: WeatherApi
        get() {
            val retrofit = Retrofit.Builder()
                    .baseUrl(BASEURL)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build()

            return retrofit.create(WeatherApi::class.java)
        }
}
